LiteCore/Storage/SQLiteDataFile.cc

```diff
@@ -600,37 +606,62 @@ namespace litecore {
     }
 
 
+    uint64_t SQLiteDataFile::fileSize() {
+        // Move all WAL changes into the main database file, so its size is accurate:
+        _exec("PRAGMA wal_checkpoint(FULL)");
+        return DataFile::fileSize();
+    }
+
+
     void SQLiteDataFile::optimize() {
-        bool logged = false;
-        if (SQL.willLog(LogLevel::Verbose)) {
-            // Log the details of what the optimize will do, before actually doing it:
-            SQLite::Statement stmt(*_sqlDb, "PRAGMA optimize(3)");
-            while (stmt.executeStep()) {
-                LogVerbose(SQL, "PRAGMA optimize ... %s", stmt.getColumn(0).getString().c_str());
-                logged = true;
+        try {
+            bool logged = false;
+            if (SQL.willLog(LogLevel::Verbose)) {
+                // Log the details of what the optimize will do, before actually doing it:
+                SQLite::Statement stmt(*_sqlDb, "PRAGMA optimize(3)");
+                while (stmt.executeStep()) {
+                    LogVerbose(SQL, "PRAGMA optimize ... %s", stmt.getColumn(0).getString().c_str());
+                    logged = true;
+                }
             }
+            if (!logged)
+                LogVerbose(SQL, "PRAGMA optimize");
```

Comment:
This is never going to be hit I think.  If the level is verbose, the previous verbose line will be logged.  If the level is not verbose, this is a no-op.  Should this be info level?

Reply:
@snej Is there a path I am missing here?

Reply:
> the previous verbose line will be logged

Not if `executeStep` never returns true, which in my experience happens most of the time.

Reply:
(anyway, that logic isn't new; it's been around a long time. It just looks new in the diff.)